File: app/src/main/java/org/michaelbel/tjgram/utils/date/DateUtil.java
package org.michaelbel.tjgram.utils.date;

import android.text.format.Time;

@SuppressWarnings("all")
public class DateUtil {

    public static boolean isMonthNow(long when) {
        Time time = new Time();
        time.set(when);

        int thenYear = time.year;
        int thenMonth = time.month;

        time.set(System.currentTimeMillis());
        return (thenYear == time.year) && (thenMonth == time.month);
    }

    public static boolean isYearNow(long when) {
        Time time = new Time();
        time.set(when);

        int thenYear = time.year;

        time.set(System.currentTimeMillis());
        return thenYear == time.year;
    }

    public static boolean isYearLast(long when) {
        Time time = new Time();
        time.set(when);

        int thenYear = time.year;

        time.set(System.currentTimeMillis());
        return thenYear < time.year;
    }
}

File: app/src/main/java/org/michaelbel/tjgram/utils/date/DateUtil.kt
package org.michaelbel.tjgram.utils.date

import android.text.format.Time

object DateUtil {

    fun isYearNow(then: Long): Boolean {
        val time = Time()
        time.set(then)

        val thenYear = time.year

        time.set(System.currentTimeMillis())
        return thenYear == time.year
    }
}